hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,266 @@
+        try:
+            subprocess.check_call(args_1)
+            subprocess.check_call(args_2)
+        except subprocess.CalledProcessError as e:
+            logger.error('%s', e)
+            raise SwitchError
+        
+#        retcode_1 = self.ovs_connect(shell_cmd_1)
+#        retcode_2 = self.ovs_connect(shell_cmd_2)
+#        if(retcode_1 is None and retcode_2 is None):
+#            return None
+
+    def modify_port(self, port, channel, new_network):
+        """ Changes vlan assignment to the port.
+
+        Args:
+            port: switch port in a valid format
+            channel: eg 'vlan/native' or 'vlan/<vlan_id>
+            new_network: vlan_id
+        Returns: If successful returns None else error.
+        """
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                return self._remove_native_vlan(interface)
+            else:
+                return self._set_native_vlan(interface, new_network)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                " switch!"
+            vlan_id = match.groups()[0]
+
+            if new_network is None:
+                return self._remove_vlan_from_port(interface, vlan_id)
+            else:
+                assert new_network == vlan_id
+                return self._add_vlan_to_trunk(interface, vlan_id)
+
+# 2. Private Methods:
+
+    def _string_to_list(self, a_string):
+        """Converts a string representation of list to list.
+        Args:
+            a_string: list output recieved as string.
+        Returns: object of list type.
+                 Empty list is put as None.
+        """
+        if a_string[0] == '[' and a_string[1] == ']':
+            return None
+        else:
+            a_string = a_string.replace("[", "['").replace("]", "']")
+            a_string = a_string.replace(",", "','")
+            a_list = ast.literal_eval(a_string)
+            a_list = [ele.strip() for ele in a_list]
+            return a_list
+
+    def _string_to_dict(self, a_string):
+        """Converts a string representation of dictionary
+        into a dictionary type object.
+
+        Args:
+            a_string: dictionary recieved as type string
+        Returns: Object of dictionary type.
+        """
+        if a_string[0] == '{' and a_string[1] == '}':
+            a_dict = ast.literal_eval(a_string)
+            return a_dict
+        elif a_string[0] == '{' and len(a_string) > 2:
+            a_string = a_string.replace("{", "{'").replace("}", "'}")
+            a_string = a_string.replace(":", "':'").replace(",", "','")
+            a_dict = ast.literal_eval(a_string)
+            a_dict = {k.strip(): v.strip() for k, v in a_dict.iteritems()}
+            return a_dict
+        else:
+            return None
```

Comment:
the only time this would happen is if you pass something other than a dictionary to this function, in that case you should raise an error rather than just returning None. What do you think?

Reply:
I am sure I pointed these out earlier.

Reply:
The dependency was different. Changed the code in `_interface_info(self, port)` which is the only consumer to this functions. removed the extra conditional statement as suggested.